ApplicationCode/ReservoirDataModel/RigGridBase.cpp

```diff
@@ -216,6 +216,25 @@ bool RigGridBase::ijkFromCellIndex( size_t cellIndex, size_t* i, size_t* j, size
     return true;
 }
 
+//--------------------------------------------------------------------------------------------------
+/// This version does no if-guarding. Check that all dimensions of the grid are non-zero before using.
+/// Useful for running in a loop after doing the sanity check once.
+//--------------------------------------------------------------------------------------------------
+void RigGridBase::ijkFromCellIndexNonGuarded( size_t cellIndex, size_t* i, size_t* j, size_t* k ) const
+{
+    size_t index = cellIndex;
+
+    size_t cellCountI = m_gridPointDimensions[0] - 1;
+    size_t cellCountJ = m_gridPointDimensions[1] - 1;
+    size_t cellCountK = m_gridPointDimensions[2] - 1;
+
+    *k = index / ( cellCountI * cellCountJ );
+    index -= ( *k ) * ( cellCountI * cellCountJ );
+    *j = index / cellCountI;
+    index -= ( *j ) * cellCountI;
+    *i = index;
```

Comment:
Just for your information, you can schedule this calculation in the reverse direction to get code that looks like
```C++
*i = index % cellCountI;  index /= cellCountI;
*j = index % cellCountJ;
*k = index / cellCountJ;
```
Whether this is an improvement is of course in the eye of the beholder.

Reply:
It is a good point. It actually allows us to use std::div as well . Which may do the % and the / in a single operation. The pull request was simply getting rid of if-statements that we may as well do once and for all and not for every single iteration in a loop.

Reply:
I may add your suggestion in a follow up commit.

Reply:
I also see I calculate cellCountK unnecessarily.

Reply:
> It actually allows us to use `std::div` as well . Which may do the `%` and the `/` in a single operation.

Using `std::div` is good in order to explicitly communicate the intent to the reader, but I don't think you actually need it to get the "single instruction" behaviour&mdash;at least not on x86_64.  Matt Godbolt's compiler explorer (https://godbolt.org/z/SdLjAQ) suggests that even the naive implementation I posted effectively boils dow to two `div` instructions at the assembly level.